e2e/main_test.go

```diff
@@ -27,20 +41,58 @@ func TestMain(m *testing.M) {
 	if err != nil {
 		panic(err)
 	}
+	dockerApp := os.Getenv("DOCKERAPP_BINARY")
 	if dockerApp == "" {
 		dockerApp = filepath.Join(cwd, "../bin/docker-app")
 	}
 	dockerApp, err = filepath.Abs(dockerApp)
 	if err != nil {
 		panic(err)
 	}
-	if dockerCli == "" {
-		dockerCli = "docker"
+	// Prepare docker cli to call the docker-app plugin binary:
+	// - Create a config dir with a custom config file
+	// - Create a symbolic link with the dockerApp binary to the plugin directory
+	if dockerCliPath == "" {
+		dockerCliPath = "docker"
+	} else {
+		dockerCliPath, err = filepath.Abs(dockerCliPath)
+		if err != nil {
+			panic(err)
+		}
+	}
+	configDir, err := ioutil.TempDir("", "config")
+	if err != nil {
+		panic(err.Error())
+	}
+	defer os.RemoveAll(configDir)
+
+	err = os.Setenv("DOCKER_CONFIG", configDir)
+	if err != nil {
+		panic(err.Error())
 	}
-	cmd := exec.Command(dockerApp, "version")
+	dockerCli = dockerCliCommand{path: dockerCliPath, config: configDir}
+
+	config := dockerConfigFile.ConfigFile{CLIPluginsExtraDirs: []string{configDir}}
+	configFile, err := os.Create(filepath.Join(configDir, "config.json"))
+	if err != nil {
+		panic(err.Error())
+	}
+	err = json.NewEncoder(configFile).Encode(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	dockerAppExecName := "docker-app"
+	if runtime.GOOS == "windows" {
+		dockerAppExecName += ".exe"
+	}
+	if err := os.Symlink(dockerApp, filepath.Join(configDir, dockerAppExecName)); err != nil {
+		panic(err.Error())
+	}
+
+	cmd := exec.Command(dockerApp, "app", "version")
 	output, err := cmd.CombinedOutput()
 	if err != nil {
-		panic(err)
+		panic(err.Error())
```

Comment:
Can we not just do a `panic(err)` here?

Reply:
I'll turn `panic(err)` uniform through the file.